src/openforms/js/components/admin/form_design/SubmissionFields.js

```diff
@@ -0,0 +1,112 @@
+import PropTypes from 'prop-types';
+import {useContext, useEffect, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {APIContext} from 'components/admin/form_design/Context';
+import ActionButton from 'components/admin/forms/ActionButton';
+import Field from 'components/admin/forms/Field';
+import Fieldset from 'components/admin/forms/Fieldset';
+import FormRow from 'components/admin/forms/FormRow';
+import {NumberInput} from 'components/admin/forms/Inputs';
+import {FormException} from 'utils/exception';
+import {patch} from 'utils/fetch';
+
+import useConfirm from './useConfirm';
+
+export const SubmissionLimitFields = ({submissionMaximumAllowed, formUuid, onChange}) => {
+  const intl = useIntl();
+  const {csrftoken} = useContext(APIContext);
+  const {ConfirmationModal, confirmationModalProps, openConfirmationModal} = useConfirm();
+  const [confirmationResult, setConfirmationResult] = useState(null);
+
+  const handleChange = event => {
+    const {name, value: initialValue} = event.target;
+    // the backend must receive a value or null since it's a nullable integer field
+    const value = initialValue === '' ? null : initialValue;
+    const updatedEvent = {...event, target: {...event.target, name, value}};
+    onChange(updatedEvent);
+  };
+
+  const resetCounter = async () => {
+    try {
+      const resetResult = await patch(
+        `/api/v2/forms/${formUuid}/reset_submission_counter`,
+        csrftoken,
+        {},
+        true
+      );
+      if (!resetResult.ok) {
+        throw new FormException(
+          'An error occurred while trying to reset the counter.',
+          resetResult.data
+        );
+      }
+    } catch (e) {
+      return null;
+    }
+    return resetResult.data;
+  };
+
+  useEffect(() => {
+    if (confirmationResult === true) {
+      resetCounter();
+    }
+  }, [confirmationResult]);
```

Comment:
don't use effects to call API endpoints because of user actions (like clicking a button), instead, perform the side effect directly as part of the `onClick` handler. Effects only run on the next render and are harder to debug/follow what's happening.

This also removes the artificial state for `confirmationResult` and generally makes the implementation simpler.